### ISSUE TITLE
Protect Mirage memory from the payload

### ISSUE THREAD

**Issue by CharlyCst** [CharlyCst/miralis#12]
Currently Mirage does not protect its own memory from the payload. We want to add support for memory protection to enforce isolation between Mirage and its payload.

---

TODO: decide on an implementation strategy

- Should we add support for lock bits? Most likely not at first, create a new issue to track the need of doing so in the future.
- How to decide on accessible memory for the virtual firmware? For now let's grant access to everything except mirage, in the future we will need to protect the OS too.
- This issue depends on #11 to test the access fault in the payload.
